File: apps/backend/generator-service/src/domain/entity/result.py
from domain.entity.base import Base
from sqlalchemy import Column, String, Integer, ForeignKey
from sqlalchemy.dialects.postgresql import UUID

class Result(Base):
    __tablename__ = 'results'
    id = Column(Integer, primary_key=True, autoincrement=True)

    dateidea_id = Column(Integer)  # Acts like a soft pointer

    job_id = Column(UUID(as_uuid=True), ForeignKey('jobs.id'))

File: apps/backend/api/app/domain/entity/result.go
package entity

import "github.com/google/uuid"

type Result struct {
	ID         uint      `gorm:"primaryKey;autoIncrement`
	DateIdeaID int       `gorm:"column:dateidea_id"` // soft pointer to DateIdea
	JobID      uuid.UUID `gorm:"column:job_id"`      // FK to jobs table
}
